File: apps/web/ios/App/App/AppsFlyerConfig.swift
import Foundation

/// Helper to read AppsFlyer configuration from Info.plist
/// Keys: AppsFlyerDevKey, AppsFlyerAppID
enum AppsFlyerConfig {
    static var devKey: String {
        return Bundle.main.object(forInfoDictionaryKey: "AppsFlyerDevKey") as? String ?? ""
    }

    static var appId: String {
        return Bundle.main.object(forInfoDictionaryKey: "AppsFlyerAppID") as? String ?? ""
    }

    static var isConfigured: Bool {
        return !devKey.isEmpty
            && !appId.isEmpty
            && !devKey.hasPrefix("__REPLACE")
            && !appId.hasPrefix("__REPLACE")
    }

    /// Log configuration status (masks sensitive devKey)
    static func logStatus() {
        let maskedKey = devKey.isEmpty ? "(empty)" : "\(devKey.prefix(6))..."
        print("[AppsFlyer] Config status:")
        print("[AppsFlyer]   devKey: \(maskedKey)")
        print("[AppsFlyer]   appId: \(appId.isEmpty ? "(empty)" : appId)")
        print("[AppsFlyer]   isConfigured: \(isConfigured)")
    }
}

File: apps/web/android/app/src/main/java/com/fitjourney/app/AppsFlyerConfig.kt
package com.fitjourney.app

import android.util.Log

/**
 * Helper to read AppsFlyer configuration from BuildConfig
 * Mirrors iOS implementation in AppsFlyerConfig.swift
 *
 * To set the key, add to your gradle.properties (or local.properties for local dev):
 *   APPSFLYER_DEV_KEY=your_appsflyer_dev_key_here
 */
object AppsFlyerConfig {
    private const val TAG = "AppsFlyer"

    val devKey: String
        get() = BuildConfig.APPSFLYER_DEV_KEY

    val isConfigured: Boolean
        get() = devKey.isNotEmpty() &&
                !devKey.startsWith("__REPLACE")

    /**
     * Log configuration status (masks sensitive devKey)
     */
    fun logStatus() {
        val maskedKey = if (devKey.isEmpty()) "(empty)" else "${devKey.take(6)}..."
        Log.d(TAG, "Config status:")
        Log.d(TAG, "  devKey: $maskedKey")
        Log.d(TAG, "  isConfigured: $isConfigured")
    }
}
